server/storage/fs.go

```diff
@@ -12,6 +12,7 @@ import (
 
 type FileSystem interface {
 	Stat(path string) (stat FileStat, err error)
+	EnsurePath(path string) (filePath string, err error)
```

Comment:
i could like to remove the `ensurePath` method, if you are using local FS as the fs, you should ensure the dir in `WriteFile` method

Reply:
for example, s3 doesn't have an actual path concept(it's just a key)

Reply:
I think the network delay will increase the result response times if we use the file storage instead of the local disk. For example s3/ GCS, but only the r2 is fast in Cloudflare.